app/Services/Backups/InitiateBackupService.php

```diff
@@ -111,6 +115,18 @@ public function handle(Server $server, string $name = null): Backup
             );
         }
 
+        // Check if the server has reached or exceeded it's backup limit
+        if (!$server->backup_limit || $server->backups()->where('is_successful', true)->count() >= $server->backup_limit) {
+            // Do not allow the user to continue if this server is already at its limit and can't override.
+            if (!$override || $server->backup_limit <= 0) {
+                throw new TooManyBackupsException($server->backup_limit);
+            }
+
+            // Remove latest backup
```

Comment:
Is this actually the latest backup or is it the oldest backup?

Reply:
Oh right. Sorry.